packages/server/src/arpa_reporter/services/persist-upload.js

```diff
@@ -152,46 +155,51 @@ async function ensureValidReportingPeriodId(reportingPeriodId) {
 async function persistUpload({
     filename, user, buffer, body,
 }) {
-    // Fetch reportingPeriodId, agencyId, and notes from the body
-    // and rename with 'supplied' prefix. These may be null.
-    const {
-        reportingPeriodId: suppliedReportingPeriodId,
-        agencyId: suppliedAgencyId,
-        notes: suppliedNotes,
-    } = body;
-
-    // Make sure we can actually read the supplied buffer (it's a valid spreadsheet)
-    await validateBuffer(buffer);
-
-    // Either use supplied reportingPeriodId,
-    // or fall back to the current reporting period ID if undefined
-    const validatedReportingPeriodId = await ensureValidReportingPeriodId(suppliedReportingPeriodId);
-
-    // Check if the user is affiliated with the given agency,
-    // or leave undefined (we'll derive it later from the spreadsheet)
-    const validatedAgencyId = await ensureValidAgencyId(suppliedAgencyId, user.id);
-
-    // Escape note text
-    const validatedNotes = ensureValidNotes(suppliedNotes);
-
-    // Create the upload row
-    const uploadData = {
-        filename,
-        reportingPeriodId: validatedReportingPeriodId,
-        userId: user.id,
-        agencyId: validatedAgencyId,
-        notes: validatedNotes,
-    };
-    const uploadRow = createUploadRow(uploadData);
-
-    // Insert the upload row into the database
-    const upload = await createUpload(uploadRow);
-
-    // Persist the upload to the filesystem
-    await persistUploadToFS(upload, buffer);
-
-    // Return the upload we created
-    return upload;
+    return tracer.trace(
```

Comment:
Do you need to `await` this call? I was looking at https://datadoghq.dev/dd-trace-js/ and https://docs.datadoghq.com/tracing/trace_collection/custom_instrumentation/nodejs/?tab=asyncawait#creating-spans

Reply:
Ahh, I guess awaiting is just a stylistic choice here.

JavaScript is actually pretty forgiving of how you use and return promises in `async` functions.  All of the following behave like functions that return a promise which resolves with the value `1`:


```
function promiseOne() {
    return Promise.resolve(1);
}

async function promiseOne() {
    return 1;
}

async function promiseOne() {
    return Promise.resolve(1);
}

async function promiseOne() {
    return await Promise.resolve(1);
}
```

Our linter is currently configured to complain about the use of a "redundant await" here if I include it, so I left that out.

<img width="765" alt="image" src="https://github.com/usdigitalresponse/usdr-gost/assets/11449340/22629ee2-e58f-4f9f-819e-1234cf33300d">

Reply:
I learn something new about Javascript every day.

Reply:
@ajhyndman I'm receiving this error in staging- I suspect it has to do with these changes potentially. 
![image](https://github.com/usdigitalresponse/usdr-gost/assets/6497366/40eee4fe-a69d-4856-841c-eae4ac9cf7db)